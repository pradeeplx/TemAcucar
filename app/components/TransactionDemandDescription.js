import React from 'react'
import { View } from 'react-native'
import { fontFactor } from "../helpers"
import truncate from 'truncate'
import Colors from "../Colors"
import Sentence from "./Sentence"
import UserImage from "./UserImage"

export default TransactionDemandDescription = ({ index, demand: { user, name }, currentUser}) => (
  <View style={{
    flexDirection: 'row',
    alignItems: 'center',
  }}>
    <UserImage source={{uri: user.image_url}} style={{marginRight: 6}} />
    <View style={{
      flexDirection: 'column',
      flex: 2,
    }}>
      <Sentence style={{
        color: Colors.darkGray,
        fontSize: 10 * fontFactor(),
      }}>
        {currentUser.id === user.id ? 'VocÃª' : user.first_name} pediu um(a)
      </Sentence>
      <Sentence style={{
        color: Colors.darkGray,
      }}>
        {truncate(name, 25)}
      </Sentence>
    </View>
  </View>
)
