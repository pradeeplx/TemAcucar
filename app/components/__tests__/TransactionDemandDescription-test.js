import 'react-native'
import React from 'react'
import TransactionDemandDescription from '../TransactionDemandDescription'
import renderer from 'react-test-renderer'

const props = {
  index: 0,
  currentUser: { id: 1 },
  demand: {
    user: {
      id: 1,
      image_url: 'http://google.com',
      first_name: 'foo',
    },
    name: 'bar',
  },
}

it('renders correctly', () => {
  const tree = renderer.create(
    <TransactionDemandDescription {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('truncates the name', () => {
  const tree = renderer.create(
    <TransactionDemandDescription {...props} demand={{ ...props.demand, name: 'foobarfoobarfoobarfoobarfoobarfoobarfoobar' }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders user name when is not currentUser', () => {
  const tree = renderer.create(
    <TransactionDemandDescription {...props} currentUser={{ id: 2 }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
