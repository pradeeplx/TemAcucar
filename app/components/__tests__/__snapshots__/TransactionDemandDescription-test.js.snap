exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }>
  <Image
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 27,
          "height": 54,
          "width": 54,
        },
        Object {
          "marginRight": 6,
        },
      ]
    } />
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "column",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
            "fontSize": 15,
          },
        ]
      }>
      Você
       pediu um(a)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      bar
    </Text>
  </View>
</View>
`;

exports[`test renders user name when is not currentUser 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }>
  <Image
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 27,
          "height": 54,
          "width": 54,
        },
        Object {
          "marginRight": 6,
        },
      ]
    } />
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "column",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
            "fontSize": 15,
          },
        ]
      }>
      foo
       pediu um(a)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      bar
    </Text>
  </View>
</View>
`;

exports[`test truncates the name 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }>
  <Image
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 27,
          "height": 54,
          "width": 54,
        },
        Object {
          "marginRight": 6,
        },
      ]
    } />
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "column",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
            "fontSize": 15,
          },
        ]
      }>
      Você
       pediu um(a)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      foobarfoobarfoobarfoob...
    </Text>
  </View>
</View>
`;
